File: 26 Remove Duplicates from Sorted Array/26.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums or len(nums) <= 1:
            return len(nums)
        slow, fast = 1, 1
        while fast < len(nums):
            if slow >= 1 and nums[fast] == nums[slow - 1]:
                fast += 1
            else:
                nums[slow] = nums[fast]
                slow += 1
                fast += 1
        return slow

File: 26 Remove Duplicates from Sorted Array/26.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length <= 1) {
            return nums.length;
        }
        int slow = 1, fast = 1;
        while (fast < nums.length) {
            if (slow >= 1 && nums[fast] == nums[slow - 1]) {
                fast++;
            } else {
                nums[slow++] = nums[fast++];
            }
        }
        return slow;
    }

    // Time complexity: O(n)
    // Space complexity: O(1)
}